shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -195,10 +187,63 @@ - (NSObject*)valuePublishedByPlugin:(NSString*)pluginKey {
   return nil;
 }
 
-#pragma mark - FlutterAppLifeCycleProvider methods
+#pragma mark - Selectors handling
 
 - (void)addApplicationLifeCycleDelegate:(NSObject<FlutterPlugin>*)delegate {
   [_lifeCycleDelegate addDelegate:delegate];
 }
 
+#pragma mark - UIApplicationDelegate method dynamic implementation
+
+- (BOOL)respondsToSelector:(SEL)selector {
+  if ([_lifeCycleDelegate isSelectorAddedDynamically:selector]) {
+    return [self delegateRespondsSelectorToPlugins:selector];
+  }
+  return [super respondsToSelector:selector];
+}
+
+- (BOOL)delegateRespondsSelectorToPlugins:(SEL)selector {
+  if ([_lifeCycleDelegate hasPluginThatRespondsToSelector:selector]) {
+    return [_lifeCycleDelegate respondsToSelector:selector];
+  } else {
+    return NO;
+  }
+}
+
+- (id)forwardingTargetForSelector:(SEL)aSelector {
+  if ([_lifeCycleDelegate isSelectorAddedDynamically:aSelector]) {
+    [self logCapabilityConfigurationWarningIfNeeded:aSelector];
+    return _lifeCycleDelegate;
+  }
+  return [super forwardingTargetForSelector:aSelector];
+}
+
+// Mimic the logging from Apple when the capability is not set for the selectors.
+// However the difference is that Apple logs these message when the app launches, we only
+// log it when the method is invoked. We can possibly also log it when the app launches, but
+// it will cause an additional scan over all the plugins.
+- (void)logCapabilityConfigurationWarningIfNeeded:(SEL)selector {
+  NSArray* backgroundModesArray =
+      [[NSBundle mainBundle] objectForInfoDictionaryKey:kUIBackgroundMode];
+  NSSet* backgroundModesSet = [[NSSet alloc] initWithArray:backgroundModesArray];
```

Comment:
Nit: Just autorelease this instead of adding the release later. Just in case we refactor this to add an early return or something.